src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Field.cs

```diff
@@ -12,11 +11,73 @@ public override BoundNode VisitFieldAccess(BoundFieldAccess node)
         {
             BoundExpression rewrittenReceiver = VisitExpression(node.ReceiverOpt);
 
-            // field access of a tuple actually accesses the underlying field.
-            FieldSymbol field = (node.FieldSymbol as TupleFieldSymbol)?.UnderlyingTypleFieldSymbol ??
-                                node.FieldSymbol;
+            var tupleField = node.FieldSymbol as TupleFieldSymbol;
+            if (tupleField != null)
+            {
+                return RewriteTupleFieldAccess(node, rewrittenReceiver);
+            }
+            else
+            {
+                return MakeFieldAccess(node.Syntax, rewrittenReceiver, node.FieldSymbol, node.ConstantValue, node.ResultKind, node.Type, node);
+            }
+        }
+
+        /// <summary>
+        /// Converts access to a tuple instance into access into the underlying ValueTuple(s).
+        ///
+        /// For instance, tuple.Item8
+        /// produces fieldAccess(field=Item1, receiver=fieldAccess(field=Rest, receiver=ValueTuple for tuple))
+        /// </summary>
+        private BoundExpression RewriteTupleFieldAccess(BoundFieldAccess node, BoundExpression rewrittenReceiver)
+        {
+            var tupleField = (TupleFieldSymbol)node.FieldSymbol;
+            var tupleType = (TupleTypeSymbol)tupleField.ContainingSymbol;
+            NamedTypeSymbol underlyingTupleType = tupleType.UnderlyingTupleType;
+
+            int fieldRemainder;
+            int fieldChainLength = TupleTypeSymbol.NumberOfValueTuples(tupleField.Position, out fieldRemainder);
+
+            BoundExpression currentField = rewrittenReceiver;
```

Comment:
The name of this local is misleading, in fact this local is not needed, use rewrittenReceiver in its place.